azure/iac/storageaccounts.rego

```diff
@@ -726,3 +726,348 @@ storage_account_latest_tls_configured_metadata := {
+}
+
+azure_issue["storage_account_private_endpoint"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.network/privateendpoints"
+    privateLinkServiceConnection := resource.properties.privateLinkServiceConnections[_]
+    contains(lower(privateLinkServiceConnection.properties.privateLinkServiceId), "microsoft.storage/storageaccounts")
+}
+
+source_path[{"storage_account_private_endpoint":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.network/privateendpoints"
+    privateLinkServiceConnection := resource.properties.privateLinkServiceConnections[j]
+    contains(lower(privateLinkServiceConnection.properties.privateLinkServiceId), "microsoft.storage/storageaccounts")
+    metadata:= {
+        "resource_path": [["resources",i,"properties","privateLinkServiceConnections",j,"properties","privateLinkServiceId"]]
+    }
+}
+
+storage_account_private_endpoint {
+	lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    azure_issue["storage_account_private_endpoint"]
+    not azure_attribute_absence["storage_account_private_endpoint"]
+}
+
+storage_account_private_endpoint = false {
+	lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    not azure_issue["storage_account_private_endpoint"]
+}
+
+storage_account_private_endpoint = false {
+	lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    azure_attribute_absence["storage_account_private_endpoint"]
+}
+
+storage_account_private_endpoint_err = "Azure Storage Account does not configure with private endpoints" {
+	lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    not azure_issue["storage_account_private_endpoint"]
+} else = "Azure Private endpoints resoruce is missing" {
+	lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    azure_attribute_absence["storage_account_private_endpoint"]
+}
+
+storage_account_private_endpoint_metadata := {
+    "Policy Code": "PR-AZR-ARM-STR-019",
+    "Type": "IaC",  
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Storage accounts should use private link",
+    "Policy Description": "Azure Private Link lets you connect your virtual network to Azure services without a public IP address at the source or destination. The Private Link platform handles the connectivity between the consumer and services over the Azure backbone network. By mapping private endpoints to your storage account, data leakage risks are reduced. Learn more about private links at - https://aka.ms/azureprivatelinkoverview",
+    "Resource Type": "microsoft.storage/storageaccounts",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts"
+}
+
+
+#
+# PR-AZR-ARM-STR-020
+#
+
+default storage_account_require_encryption = null
+
```

Comment:
@shahinshirvani add an attribute checking for `resource.properties.encryption`

Reply:
```not obj.foo.bar.baz``` checks if path foo.bar.baz, foo.bar, or foo does not exist or is false

Re:https://www.openpolicyagent.org/docs/latest/policy-reference/